payment/src/main/java/org/killbill/billing/payment/core/ProcessorBase.java

```diff
@@ -204,16 +215,32 @@ public WithAccountLock(final PaymentConfig paymentConfig) {
         }
     }
 
-    protected static <ReturnType> ReturnType dispatchWithExceptionHandling(@Nullable final Account account, final Callable<PluginDispatcherReturnType<ReturnType>> callable, PluginDispatcher<ReturnType> pluginFormDispatcher) throws PaymentApiException {
+    protected static <ReturnType> ReturnType dispatchWithExceptionHandling(@Nullable final Account account, final String pluginName, final Callable<PluginDispatcherReturnType<ReturnType>> callable, PluginDispatcher<ReturnType> pluginFormDispatcher) throws PaymentApiException {
         final UUID accountId = account != null ? account.getId() : null;
         final String accountExternalKey = account != null ? account.getExternalKey() : "";
+
+        final RequestData requestData = Request.getPerThreadRequestData();
+        final String requestId;
+        if (requestData != null) {
+            requestId = requestData.getRequestId();
+        } else {
+            requestId = "notAvailabeRequestId";
+        }
+
         try {
-            return pluginFormDispatcher.dispatchWithTimeout(callable);
+            log.debug("Calling plugin {} with requestId {}", pluginName, requestId);
+            ReturnType result = pluginFormDispatcher.dispatchWithTimeout(callable);
+            log.debug("Successful call of plugin {} for account {} with result {} and requestId {}", pluginName, account.getExternalKey(), result, requestId);
+            return result;
         } catch (final TimeoutException e) {
-            throw new PaymentApiException(ErrorCode.PAYMENT_PLUGIN_TIMEOUT, accountId, null);
+            final String errorMessage = String.format("TimeoutException during the execution of plugin %s with requestId %s ", pluginName, requestId);
+            log.error(errorMessage, e);
+            throw new PaymentApiException(ErrorCode.PAYMENT_PLUGIN_TIMEOUT, accountId, errorMessage);
         } catch (final InterruptedException e) {
             Thread.currentThread().interrupt();
-            throw new PaymentApiException(ErrorCode.PAYMENT_INTERNAL_ERROR, Objects.firstNonNull(e.getMessage(), ""));
+            final String errorMessage = String.format("InterruptedException during the execution of plugin %s with requestId %s ", pluginName, requestId);
+            log.error(errorMessage, e);
+            throw new PaymentApiException(ErrorCode.PAYMENT_INTERNAL_ERROR, Objects.firstNonNull(e.getMessage(), errorMessage));
         } catch (final ExecutionException e) {
```

Comment:
The idea to log the request id with the exception just makes sense in the connection with the logging messages that surround the actual plugin call. Especially in callables that contain many plugin calls you can investigate with the request id for which plugin a log statement is missing that says the the plugin has been called successfully. In this case just one plugin is called, but I added it here for consistency reason. I like the idea to add the request id to the logging in general, because our experience with GPS shows that it is really helpful for debugging.